### ISSUE TITLE
Matrix url parameters break route matching

### ISSUE THREAD

**Issue by dbpieter** [single-spa/single-spa-layout#182]
I'm currently configuring our existing single-spa app to use layout-engine.

`<single-spa-router>
  <application name="navigation"></application>

  <route path="some-mfe">
    <application name="some-mfe"></application>
  </route>
</single-spa-router>`

When navigating to mydomain.com/some-mfe I see the app being bootstrapped and briefly appearing. However as soon as any matrix parameters are added to the url (ex mydomain.com/some-mfe;pageSize=50;pageNumber=8) the app is unmounted by single-spa. Matrix parameters are commonly used in Angular as routing parameters. 
It seems that the route matching algorithm/regex doesn't like the semicolon ?

**Comment by filoxo** [comment]
correct, because matrix url parameters aren't an accepted web standard regardless of Angular's usage so support hasn't been added. however, i would think it would be relatively easy to support them in [matchRoute.js](https://github.com/single-spa/single-spa-layout/blob/main/src/isomorphic/matchRoute.js) which has an accompanying [suite of tests](https://github.com/single-spa/single-spa-layout/blob/main/test/matchRoute.test.js). would the criteria be that those URL params are ignored during matching, just like search params are?
